rust/hyperlane-base/src/contract_sync/mod.rs

```diff
@@ -60,13 +64,16 @@ where
         loop {
             indexed_height.set(cursor.latest_block() as i64);
             let Ok((action, eta)) = cursor.next_action().await else {
+                sleep(Duration::from_millis(SLEEP_MS)).await;
                 continue;
             };
-            match action {
-                CursorAction::Query(range) => {
+            let sleep_duration = match action {
+                CursorAction::Query(range) => loop {
```

Comment:
using `loop` so I can break with a return value (the sleep duration)

Reply:
interesting, can you maybe add this as an in-code comment too